src/components/views/story-editor-view/story-editor-view.tsx

```diff
@@ -64,8 +68,15 @@ const StoryEditorView: React.FC<Props> = ({
     const engine = engineRef.current;
     const nodes = engine.getModel().getNodes();
 
+    if(knots[0] && knots[0].scope !== KnotScope.Global) {
+      knots[0].scope = KnotScope.Global;
+    }
+    if (knots[1] && knots[1].scope !== KnotScope.Home) {
+      knots[1].scope = KnotScope.Home;
+    }
```

Comment:
What if knots[0] is not correct global knot or knots[1] is not correct home knot? This just forces the scopes to those knots, but does not actually confirm that the knots are the correct ones. This seems it can cause major problems, if knot order is somehow different from what it should be.